<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Methods for selecting genes — genePeakSelection • Cascade</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Methods for selecting genes — genePeakSelection" />

<meta property="og:description" content="Selection of differentially expressed genes." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Cascade</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="https://cran.r-project.org/web/packages/Cascade/vignettes/Cascade.pdf">Introduction to the Cascade package and application to the GSE39411 dataset</a>
    </li>
    <li>
      <a href="https://cran.r-project.org/web/packages/Cascade/vignettes/E-MTAB-1475_re-analysis.pdf">Additional application of the Cascade package to E-MTAB-1475 dataset</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fbertran/Cascade">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Methods for selecting genes</h1>
    
    <div class="hidden name"><code>genePeakSelection-methods.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Selection of differentially expressed genes.</p>
    
    </div>

    <pre class="usage"><span class='fu'>geneSelection</span>(<span class='no'>x</span>,<span class='no'>y</span>,<span class='no'>tot.number</span>,<span class='no'>...</span>)
        <span class='fu'>genePeakSelection</span>(<span class='no'>x</span>,<span class='no'>peak</span>,<span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>either a micro_array object or a list of micro_array objects. In the first case, the micro_array object represents the stimulated measurements. In the second case, the control unstimulated data (if present) should be the first element of the list.</p></td>
    </tr>
    <tr>
      <th>y</th>
      <td><p>either a micro_array object or a list of strings. In the first case, the micro_array object represents the stimulated measurements. In the second case, the list is the way to specify the contrast:</p><dl class='dl-horizontal'>
<dt>First element:</dt><dd><p>condition, condition&amp;time or pattern. The condition specification is used when the overall is to compare two conditions. 
The condition&amp;time specification is used when comparing two conditions at two precise time points.
The pattern specification allows to decide which time point should be differentially expressed.</p></dd>
<dt>Second element:</dt><dd><p>a vector of length 2. The two conditions which should be compared. If a condition is used as control, it should be the first element of the vector. However, if this control is not measured throught time, 
the option cont=TRUE should be used.</p></dd>
<dt>Third element:</dt><dd><p>depends on the first element. 
It is no needed if condition has been specified. 
If condition&amp;time has been specified, then this is a vector containing the time point at which the comparison should be done.
If pattern has been specified, then this is a vector of 0 and 1 of length T, where T is the number of time points. The time points with desired differential expression are provided with 1.</p></dd></dl></td>
    </tr>
    <tr>
      <th>tot.number</th>
      <td><p>an integer. The number of selected genes. If tot.number &lt;0 all differentially genes are selected. If tot.number &gt; 1, tot.number is the maximum of diffenrtially genes that will be selected. 
 If 0&lt;tot.number&lt;1, tot.number represents the proportion of diffenrentially genes that are selected.</p></td>
    </tr>
    <tr>
      <th>peak</th>
      <td><p>interger. At which time points measurements should the genes be selected [optionnal for geneSelection].</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Optional arguments:</p><dl class='dl-horizontal'>
  <dt>M2</dt><dd><p>a micro_array object. The unstimulated measurements.</p></dd>
 		 		<dt>data_log</dt><dd><p>logical (default to TRUE) ; should data be logged ?</p></dd>
 		<dt>wanted.patterns</dt><dd><p>a matrix with wanted patterns [only for geneSelection].</p></dd>
 		<dt>forbidden.patterns</dt><dd><p>a matrix with forbidden patterns [only for geneSelection].</p></dd>
 		<dt>durPeak</dt><dd><p>vector of size 2 (default to c(1,1)) ; the first elements gives the length of the peak at the left, the second at the right. [only for genePeakSelection]</p></dd>
 		<dt>abs_val</dt><dd><p>logical (default to TRUE) ; should genes be selected on the basis of their absolute value expression ? [only for genePeakSelection]</p></dd>
 		<dt>alpha_diff</dt><dd><p>float ; the risk level</p></dd>
 </dl></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A micro_array object.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Jung, N., Bertrand, F., Bahram, S., Vallat, L., and Maumy-Bertrand, M. (2014). Cascade: a R-package to study, predict and simulate the diffusion of a signal through a temporal gene network. <em>Bioinformatics</em>, btt705.</p>	
<p>Vallat, L., Kemper, C. A., Jung, N., Maumy-Bertrand, M., Bertrand, F., Meyer, N., ... &amp; Bahram, S. (2013). Reverse-engineering the genetic circuitry of a cancer cell with predicted intervention in chronic lymphocytic leukemia. <em>Proceedings of the National Academy of Sciences</em>, 110(2), 459-464.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>  <span class='kw'>if</span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>require</a></span>(<span class='no'>CascadeData</span>)){
        <span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>micro_US</span>)
        <span class='no'>micro_US</span><span class='kw'>&lt;-</span><span class='fu'><a href='as.micro_array.html'>as.micro_array</a></span>(<span class='no'>micro_US</span>,<span class='kw'>time</span><span class='kw'>=</span><span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>60</span>,<span class='fl'>90</span>,<span class='fl'>210</span>,<span class='fl'>390</span>),<span class='kw'>subject</span><span class='kw'>=</span><span class='fl'>6</span>)
        <span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>micro_S</span>)
        <span class='no'>micro_S</span><span class='kw'>&lt;-</span><span class='fu'><a href='as.micro_array.html'>as.micro_array</a></span>(<span class='no'>micro_S</span>,<span class='kw'>time</span><span class='kw'>=</span><span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>60</span>,<span class='fl'>90</span>,<span class='fl'>210</span>,<span class='fl'>390</span>),<span class='kw'>subject</span><span class='kw'>=</span><span class='fl'>6</span>)

  <span class='co'>#Basically, to find the 50 more significant expressed genes you will use:</span>
  <span class='no'>Selection_1</span><span class='kw'>&lt;-</span><span class='fu'>geneSelection</span>(<span class='kw'>x</span><span class='kw'>=</span><span class='no'>micro_S</span>,<span class='kw'>y</span><span class='kw'>=</span><span class='no'>micro_US</span>,
  <span class='kw'>tot.number</span><span class='kw'>=</span><span class='fl'>50</span>,<span class='kw'>data_log</span><span class='kw'>=</span><span class='fl'>TRUE</span>)
  <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/summary'>summary</a></span>(<span class='no'>Selection_1</span>)

  <span class='co'>#If we want to select genes that are differentially </span>
  <span class='co'>#at time t60 or t90 :</span>
  <span class='no'>Selection_2</span><span class='kw'>&lt;-</span><span class='fu'>geneSelection</span>(<span class='kw'>x</span><span class='kw'>=</span><span class='no'>micro_S</span>,<span class='kw'>y</span><span class='kw'>=</span><span class='no'>micro_US</span>,<span class='kw'>tot.number</span><span class='kw'>=</span><span class='fl'>30</span>,
  <span class='kw'>wanted.patterns</span><span class='kw'>=</span>
  <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/cbind'>rbind</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>0</span>,<span class='fl'>1</span>,<span class='fl'>0</span>,<span class='fl'>0</span>),<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>1</span>,<span class='fl'>0</span>,<span class='fl'>0</span>,<span class='fl'>0</span>),<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>1</span>,<span class='fl'>1</span>,<span class='fl'>0</span>,<span class='fl'>0</span>)))
  <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/summary'>summary</a></span>(<span class='no'>Selection_2</span>)

  <span class='co'>#To select genes that have a differential maximum of expression at a specific time point.</span>

  <span class='no'>Selection_3</span><span class='kw'>&lt;-</span><span class='fu'>genePeakSelection</span>(<span class='kw'>x</span><span class='kw'>=</span><span class='no'>micro_S</span>,<span class='kw'>y</span><span class='kw'>=</span><span class='no'>micro_US</span>,<span class='kw'>peak</span><span class='kw'>=</span><span class='fl'>1</span>,
  <span class='kw'>abs_val</span><span class='kw'>=</span><span class='fl'>FALSE</span>,<span class='kw'>alpha_diff</span><span class='kw'>=</span><span class='fl'>0.01</span>)
  <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/summary'>summary</a></span>(<span class='no'>Selection_3</span>)
  }</div><div class='output co'>#&gt; <span class='message'>Loading required package: CascadeData</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: limma</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: cluster</span></div><div class='output co'>#&gt;       US60               US90             US210            US390        
#&gt;  Min.   :-0.44125   Min.   :-0.8100   Min.   :-1.560   Min.   :-1.0414  
#&gt;  1st Qu.:-0.08878   1st Qu.:-0.1214   1st Qu.: 1.178   1st Qu.: 0.7883  
#&gt;  Median : 0.06389   Median : 0.0170   Median : 1.537   Median : 1.2396  
#&gt;  Mean   : 0.29793   Mean   : 0.3231   Mean   : 1.509   Mean   : 1.1426  
#&gt;  3rd Qu.: 0.18056   3rd Qu.: 0.1667   3rd Qu.: 1.919   3rd Qu.: 1.5070  
#&gt;  Max.   : 2.86440   Max.   : 4.2847   Max.   : 3.673   Max.   : 4.6843  
#&gt;       US60              US90              US210             US390        
#&gt;  Min.   :-0.9303   Min.   :-0.12274   Min.   :-1.2161   Min.   :-0.9591  
#&gt;  1st Qu.:-0.4365   1st Qu.: 0.05945   1st Qu.: 0.8696   1st Qu.: 0.4383  
#&gt;  Median :-0.2202   Median : 0.15432   Median : 1.1855   Median : 0.7457  
#&gt;  Mean   :-0.1439   Mean   : 0.43625   Mean   : 1.2308   Mean   : 0.7790  
#&gt;  3rd Qu.:-0.0282   3rd Qu.: 0.29036   3rd Qu.: 1.6403   3rd Qu.: 1.0575  
#&gt;  Max.   : 2.0267   Max.   : 3.37588   Max.   : 2.6157   Max.   : 4.6113  
#&gt;       US60               US90              US210            US390       
#&gt;  Min.   :-0.62452   Min.   :-0.53921   Min.   :-1.653   Min.   :-1.611  
#&gt;  1st Qu.:-0.03412   1st Qu.:-0.03297   1st Qu.: 1.402   1st Qu.: 0.812  
#&gt;  Median : 0.06132   Median : 0.09073   Median : 1.779   Median : 1.091  
#&gt;  Mean   : 0.34579   Mean   : 0.68574   Mean   : 1.836   Mean   : 1.075  
#&gt;  3rd Qu.: 0.13643   3rd Qu.: 0.77977   3rd Qu.: 2.231   3rd Qu.: 1.412  
#&gt;  Max.   : 3.31723   Max.   : 4.36036   Max.   : 4.271   Max.   : 4.506  
#&gt;       US60               US90              US210             US390        
#&gt;  Min.   :-0.33647   Min.   :-0.46488   Min.   :-0.8332   Min.   :-0.5609  
#&gt;  1st Qu.:-0.02169   1st Qu.:-0.06483   1st Qu.: 0.7915   1st Qu.: 0.4747  
#&gt;  Median : 0.04853   Median : 0.04034   Median : 1.0525   Median : 0.6206  
#&gt;  Mean   : 0.22706   Mean   : 0.44077   Mean   : 1.1166   Mean   : 0.7000  
#&gt;  3rd Qu.: 0.18651   3rd Qu.: 0.59738   3rd Qu.: 1.4347   3rd Qu.: 0.8204  
#&gt;  Max.   : 1.82903   Max.   : 3.60640   Max.   : 2.9587   Max.   : 3.2116  
#&gt;       US60               US90              US210             US390        
#&gt;  Min.   :-0.46722   Min.   :-0.46073   Min.   :-1.0271   Min.   :-1.0395  
#&gt;  1st Qu.:-0.17848   1st Qu.:-0.02118   1st Qu.: 0.8441   1st Qu.: 0.4687  
#&gt;  Median :-0.09357   Median : 0.13574   Median : 1.2569   Median : 0.7384  
#&gt;  Mean   : 0.13903   Mean   : 0.47404   Mean   : 1.2451   Mean   : 0.7381  
#&gt;  3rd Qu.: 0.04130   3rd Qu.: 0.60663   3rd Qu.: 1.7215   3rd Qu.: 0.9835  
#&gt;  Max.   : 2.31074   Max.   : 3.24454   Max.   : 2.5446   Max.   : 3.4299  
#&gt;       US60               US90               US210            US390        
#&gt;  Min.   :-0.19230   Min.   :-0.688731   Min.   :-1.472   Min.   :-1.2069  
#&gt;  1st Qu.:-0.02128   1st Qu.: 0.009111   1st Qu.: 1.100   1st Qu.: 0.5209  
#&gt;  Median : 0.02499   Median : 0.176150   Median : 1.419   Median : 0.8359  
#&gt;  Mean   : 0.37176   Mean   : 0.566765   Mean   : 1.391   Mean   : 0.8804  
#&gt;  3rd Qu.: 0.12327   3rd Qu.: 0.828888   3rd Qu.: 1.858   3rd Qu.: 1.2084  
#&gt;  Max.   : 3.16035   Max.   : 3.199747   Max.   : 2.803   Max.   : 4.1558  </div><div class='img'><img src='genePeakSelection-methods-1.png' alt='' width='700' height='433' /></div><div class='img'><img src='genePeakSelection-methods-2.png' alt='' width='700' height='433' /></div><div class='img'><img src='genePeakSelection-methods-3.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt;       US60              US90             US210             US390         
#&gt;  Min.   :-2.5257   Min.   :-2.3695   Min.   :-2.2557   Min.   :-1.63142  
#&gt;  1st Qu.:-1.6127   1st Qu.:-0.2866   1st Qu.:-0.2105   1st Qu.:-0.20828  
#&gt;  Median :-0.3778   Median : 0.6600   Median : 0.1241   Median :-0.09562  
#&gt;  Mean   :-0.5575   Mean   : 0.3237   Mean   : 0.2928   Mean   :-0.02706  
#&gt;  3rd Qu.: 0.2975   3rd Qu.: 0.8961   3rd Qu.: 0.6911   3rd Qu.: 0.23099  
#&gt;  Max.   : 2.3763   Max.   : 2.4655   Max.   : 2.1848   Max.   : 0.78412  
#&gt;       US60              US90               US210              US390         
#&gt;  Min.   :-2.7932   Min.   :-2.492454   Min.   :-2.61740   Min.   :-1.67813  
#&gt;  1st Qu.:-0.9423   1st Qu.:-0.201708   1st Qu.:-0.26291   1st Qu.:-0.43358  
#&gt;  Median :-0.3059   Median :-0.009419   Median : 0.06493   Median : 0.00995  
#&gt;  Mean   :-0.3262   Mean   : 0.265084   Mean   :-0.02941   Mean   :-0.03069  
#&gt;  3rd Qu.: 0.1071   3rd Qu.: 0.620449   3rd Qu.: 0.28656   3rd Qu.: 0.21746  
#&gt;  Max.   : 2.2920   Max.   : 5.318566   Max.   : 1.28785   Max.   : 2.83321  
#&gt;       US60              US90             US210             US390          
#&gt;  Min.   :-2.9444   Min.   :-0.9721   Min.   :-1.5506   Min.   :-1.682560  
#&gt;  1st Qu.:-1.1363   1st Qu.:-0.3163   1st Qu.:-0.8141   1st Qu.:-0.405205  
#&gt;  Median :-0.5098   Median : 0.0250   Median :-0.1471   Median :-0.222479  
#&gt;  Mean   :-0.3967   Mean   : 0.2474   Mean   :-0.1510   Mean   :-0.246058  
#&gt;  3rd Qu.: 0.4925   3rd Qu.: 0.6905   3rd Qu.: 0.3529   3rd Qu.:-0.009658  
#&gt;  Max.   : 2.1761   Max.   : 2.5878   Max.   : 1.8377   Max.   : 0.563272  
#&gt;       US60               US90             US210              US390         
#&gt;  Min.   :-2.85438   Min.   :-0.9036   Min.   :-0.50038   Min.   :-0.89875  
#&gt;  1st Qu.:-0.43892   1st Qu.:-0.4123   1st Qu.:-0.10579   1st Qu.:-0.09241  
#&gt;  Median :-0.15123   Median : 0.1225   Median :-0.01508   Median : 0.04142  
#&gt;  Mean   :-0.18276   Mean   : 0.1603   Mean   : 0.06557   Mean   : 0.11943  
#&gt;  3rd Qu.: 0.06281   3rd Qu.: 0.6199   3rd Qu.: 0.18144   3rd Qu.: 0.18669  
#&gt;  Max.   : 1.00712   Max.   : 2.2225   Max.   : 1.12214   Max.   : 1.67398  
#&gt;       US60              US90              US210              US390         
#&gt;  Min.   :-1.3800   Min.   :-2.94444   Min.   :-1.01720   Min.   :-1.32176  
#&gt;  1st Qu.:-0.3687   1st Qu.:-0.18634   1st Qu.:-0.11358   1st Qu.:-0.35558  
#&gt;  Median :-0.1100   Median : 0.05601   Median : 0.04743   Median :-0.09259  
#&gt;  Mean   : 0.1080   Mean   : 0.02899   Mean   : 0.07287   Mean   :-0.01982  
#&gt;  3rd Qu.: 0.6869   3rd Qu.: 0.48146   3rd Qu.: 0.27237   3rd Qu.: 0.29955  
#&gt;  Max.   : 1.7346   Max.   : 1.09182   Max.   : 0.97792   Max.   : 0.84730  
#&gt;       US60              US90              US210             US390         
#&gt;  Min.   :-1.7918   Min.   :-3.20791   Min.   :-0.5665   Min.   :-0.40829  
#&gt;  1st Qu.:-0.5119   1st Qu.:-0.41715   1st Qu.:-0.0899   1st Qu.:-0.04835  
#&gt;  Median :-0.1389   Median : 0.18232   Median : 0.1129   Median : 0.18232  
#&gt;  Mean   :-0.0530   Mean   : 0.07067   Mean   : 0.2025   Mean   : 0.22579  
#&gt;  3rd Qu.: 0.5018   3rd Qu.: 0.66872   3rd Qu.: 0.6079   3rd Qu.: 0.35914  
#&gt;  Max.   : 1.9545   Max.   : 2.11718   Max.   : 0.9111   Max.   : 1.12481  </div><div class='img'><img src='genePeakSelection-methods-4.png' alt='' width='700' height='433' /></div><div class='img'><img src='genePeakSelection-methods-5.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; <span class='warning'>Warning: cannot compute exact p-value with ties</span></div><div class='output co'>#&gt; <span class='warning'>Warning: cannot compute exact p-value with ties</span></div><div class='output co'>#&gt; <span class='warning'>Warning: cannot compute exact p-value with ties</span></div><div class='output co'>#&gt; <span class='warning'>Warning: cannot compute exact p-value with ties</span></div><div class='output co'>#&gt; <span class='warning'>Warning: cannot compute exact p-value with ties</span></div><div class='output co'>#&gt; <span class='warning'>Warning: cannot compute exact p-value with ties</span></div><div class='output co'>#&gt; <span class='warning'>Warning: cannot compute exact p-value with ties</span></div><div class='output co'>#&gt; <span class='warning'>Warning: cannot compute exact p-value with ties</span></div><div class='output co'>#&gt; <span class='warning'>Warning: cannot compute exact p-value with ties</span></div><div class='output co'>#&gt; <span class='warning'>Warning: cannot compute exact p-value with ties</span></div><div class='img'><img src='genePeakSelection-methods-6.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt;       US60               US90              US210            US390        
#&gt;  Min.   :-2.08980   Min.   :-0.31237   Min.   :0.1792   Min.   :-0.1907  
#&gt;  1st Qu.:-1.97764   1st Qu.:-0.19129   1st Qu.:0.5294   1st Qu.: 0.2341  
#&gt;  Median :-0.47593   Median : 0.04929   Median :0.7023   Median : 0.3501  
#&gt;  Mean   :-0.81483   Mean   : 0.25813   Mean   :0.9934   Mean   : 0.6023  
#&gt;  3rd Qu.:-0.09126   3rd Qu.: 0.54588   3rd Qu.:1.6357   3rd Qu.: 0.9610  
#&gt;  Max.   : 0.68123   Max.   : 1.53985   Max.   :2.1467   Max.   : 1.5644  
#&gt;       US60              US90              US210             US390        
#&gt;  Min.   :-2.6684   Min.   :-0.26913   Min.   :0.06714   Min.   :-0.8210  
#&gt;  1st Qu.:-0.8209   1st Qu.:-0.01513   1st Qu.:0.33706   1st Qu.: 0.1477  
#&gt;  Median :-0.6285   Median : 0.43284   Median :0.57086   Median : 0.3322  
#&gt;  Mean   :-0.8521   Mean   : 0.59842   Mean   :1.02346   Mean   : 0.2843  
#&gt;  3rd Qu.:-0.5024   3rd Qu.: 0.81355   3rd Qu.:1.95153   3rd Qu.: 0.5248  
#&gt;  Max.   :-0.2160   Max.   : 2.52972   Max.   :2.46712   Max.   : 1.0576  
#&gt;       US60              US90              US210            US390         
#&gt;  Min.   :-1.7362   Min.   :-0.37531   Min.   :0.4137   Min.   :-0.39973  
#&gt;  1st Qu.:-1.2935   1st Qu.:-0.05243   1st Qu.:0.6403   1st Qu.: 0.01476  
#&gt;  Median :-0.6122   Median : 0.15453   Median :0.9058   Median : 0.39314  
#&gt;  Mean   :-0.6003   Mean   : 0.42682   Mean   :1.4471   Mean   : 0.60801  
#&gt;  3rd Qu.:-0.1362   3rd Qu.: 0.39540   3rd Qu.:1.7758   3rd Qu.: 1.02303  
#&gt;  Max.   : 1.0030   Max.   : 2.61726   Max.   :4.0489   Max.   : 1.98138  
#&gt;       US60              US90              US210               US390        
#&gt;  Min.   :-0.8031   Min.   :-0.37068   Min.   :-0.004577   Min.   :-0.3330  
#&gt;  1st Qu.:-0.3147   1st Qu.:-0.03856   1st Qu.: 0.380320   1st Qu.: 0.1424  
#&gt;  Median :-0.2022   Median : 0.16561   Median : 0.695225   Median : 0.1763  
#&gt;  Mean   :-0.1834   Mean   : 0.31788   Mean   : 0.845885   Mean   : 0.2404  
#&gt;  3rd Qu.:-0.1566   3rd Qu.: 0.39576   3rd Qu.: 1.079119   3rd Qu.: 0.2242  
#&gt;  Max.   : 0.7312   Max.   : 1.85850   Max.   : 2.131919   Max.   : 1.1892  
#&gt;       US60               US90             US210             US390         
#&gt;  Min.   :-1.44692   Min.   :-0.2031   Min.   :-0.2513   Min.   :-0.22907  
#&gt;  1st Qu.:-0.58974   1st Qu.: 0.1317   1st Qu.: 0.3010   1st Qu.: 0.01654  
#&gt;  Median :-0.30087   Median : 0.1696   Median : 0.4453   Median : 0.23275  
#&gt;  Mean   :-0.37780   Mean   : 0.3969   Mean   : 0.7177   Mean   : 0.31001  
#&gt;  3rd Qu.:-0.09775   3rd Qu.: 0.5669   3rd Qu.: 1.0310   3rd Qu.: 0.61957  
#&gt;  Max.   : 0.35066   Max.   : 1.4722   Max.   : 2.4299   Max.   : 1.02008  
#&gt;       US60               US90              US210            US390         
#&gt;  Min.   :-1.09861   Min.   :-0.11204   Min.   :0.3050   Min.   :-0.09353  
#&gt;  1st Qu.:-0.54020   1st Qu.: 0.04746   1st Qu.:0.5177   1st Qu.: 0.22531  
#&gt;  Median :-0.11493   Median : 0.11738   Median :0.6666   Median : 0.51500  
#&gt;  Mean   :-0.22662   Mean   : 0.33689   Mean   :1.0728   Mean   : 0.74727  
#&gt;  3rd Qu.: 0.03167   3rd Qu.: 0.21122   3rd Qu.:1.5670   3rd Qu.: 0.91661  
#&gt;  Max.   : 0.75524   Max.   : 2.05415   Max.   :2.5510   Max.   : 2.28725  </div><div class='img'><img src='genePeakSelection-methods-7.png' alt='' width='700' height='433' /></div><div class='img'><img src='genePeakSelection-methods-8.png' alt='' width='700' height='433' /></div><div class='img'><img src='genePeakSelection-methods-9.png' alt='' width='700' height='433' /></div><div class='input'>
  <span class='kw'>if</span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>require</a></span>(<span class='no'>CascadeData</span>)){
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>micro_US</span>)
<span class='no'>micro_US</span><span class='kw'>&lt;-</span><span class='fu'><a href='as.micro_array.html'>as.micro_array</a></span>(<span class='no'>micro_US</span>,<span class='kw'>time</span><span class='kw'>=</span><span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>60</span>,<span class='fl'>90</span>,<span class='fl'>210</span>,<span class='fl'>390</span>),<span class='kw'>subject</span><span class='kw'>=</span><span class='fl'>6</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>micro_S</span>)
<span class='no'>micro_S</span><span class='kw'>&lt;-</span><span class='fu'><a href='as.micro_array.html'>as.micro_array</a></span>(<span class='no'>micro_S</span>,<span class='kw'>time</span><span class='kw'>=</span><span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>60</span>,<span class='fl'>90</span>,<span class='fl'>210</span>,<span class='fl'>390</span>),<span class='kw'>subject</span><span class='kw'>=</span><span class='fl'>6</span>)
<span class='co'>#Genes with differential expression at t1</span>
<span class='no'>Selection1</span><span class='kw'>&lt;-</span><span class='fu'>geneSelection</span>(<span class='kw'>x</span><span class='kw'>=</span><span class='no'>micro_S</span>,<span class='kw'>y</span><span class='kw'>=</span><span class='no'>micro_US</span>,<span class='fl'>20</span>,<span class='kw'>wanted.patterns</span><span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/cbind'>rbind</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>1</span>,<span class='fl'>0</span>,<span class='fl'>0</span>,<span class='fl'>0</span>)))
<span class='co'>#Genes with differential expression at t2</span>
<span class='no'>Selection2</span><span class='kw'>&lt;-</span><span class='fu'>geneSelection</span>(<span class='kw'>x</span><span class='kw'>=</span><span class='no'>micro_S</span>,<span class='kw'>y</span><span class='kw'>=</span><span class='no'>micro_US</span>,<span class='fl'>20</span>,<span class='kw'>wanted.patterns</span><span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/cbind'>rbind</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>0</span>,<span class='fl'>1</span>,<span class='fl'>0</span>,<span class='fl'>0</span>)))
<span class='co'>#Genes with differential expression at t3</span>
<span class='no'>Selection3</span><span class='kw'>&lt;-</span><span class='fu'>geneSelection</span>(<span class='kw'>x</span><span class='kw'>=</span><span class='no'>micro_S</span>,<span class='kw'>y</span><span class='kw'>=</span><span class='no'>micro_US</span>,<span class='fl'>20</span>,<span class='kw'>wanted.patterns</span><span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/cbind'>rbind</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>0</span>,<span class='fl'>0</span>,<span class='fl'>1</span>,<span class='fl'>0</span>)))
<span class='co'>#Genes with differential expression at t4</span>
<span class='no'>Selection4</span><span class='kw'>&lt;-</span><span class='fu'>geneSelection</span>(<span class='kw'>x</span><span class='kw'>=</span><span class='no'>micro_S</span>,<span class='kw'>y</span><span class='kw'>=</span><span class='no'>micro_US</span>,<span class='fl'>20</span>,<span class='kw'>wanted.patterns</span><span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/cbind'>rbind</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>0</span>,<span class='fl'>0</span>,<span class='fl'>0</span>,<span class='fl'>1</span>)))
<span class='co'>#Genes with global differential expression </span>
<span class='no'>Selection5</span><span class='kw'>&lt;-</span><span class='fu'>geneSelection</span>(<span class='kw'>x</span><span class='kw'>=</span><span class='no'>micro_S</span>,<span class='kw'>y</span><span class='kw'>=</span><span class='no'>micro_US</span>,<span class='fl'>20</span>)

<span class='co'>#We then merge these selections:</span>
<span class='no'>Selection</span><span class='kw'>&lt;-</span><span class='fu'><a href='unionMicro-methods.html'>unionMicro</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/list'>list</a></span>(<span class='no'>Selection1</span>,<span class='no'>Selection2</span>,<span class='no'>Selection3</span>,<span class='no'>Selection4</span>,<span class='no'>Selection5</span>))
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/print'>print</a></span>(<span class='no'>Selection</span>)

<span class='co'>#Prints the correlation graphics Figure 4:</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/summary'>summary</a></span>(<span class='no'>Selection</span>,<span class='fl'>3</span>)

<span class='co'>##Uncomment this code to retrieve geneids.</span>
<span class='co'>#library(org.Hs.eg.db)</span>
<span class='co'>#</span>
<span class='co'>#ff&lt;-function(x){substr(x, 1, nchar(x)-3)}</span>
<span class='co'>#ff&lt;-Vectorize(ff)</span>
<span class='co'>#</span>
<span class='co'>##Here is the function to transform the probeset names to gene ID.</span>
<span class='co'>#</span>
<span class='co'>#library("hgu133plus2.db")</span>
<span class='co'>#</span>
<span class='co'>#probe_to_id&lt;-function(n){  </span>
<span class='co'>#x &lt;- hgu133plus2SYMBOL</span>
<span class='co'>#mp&lt;-mappedkeys(x)</span>
<span class='co'>#xx &lt;- unlist(as.list(x[mp]))</span>
<span class='co'>#genes_all = xx[(n)]</span>
<span class='co'>#genes_all[is.na(genes_all)]&lt;-"unknown"</span>
<span class='co'>#return(genes_all)</span>
<span class='co'>#}</span>
<span class='co'>#Selection@name&lt;-probe_to_id(Selection@name)</span>
  }</div><div class='output co'>#&gt; <span class='error'>Error in dimnames(x) &lt;- dn: 'dimnames' applied to non-array</span></div><div class='input'>	</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#references">References</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Nicolas Jung, Frédéric Bertrand , Myriam Maumy-Bertrand.</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Frederic Bertrand, Myriam Maumy-Bertrand.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

