<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Methods for selecting genes — geneSelection,micro_array,micro_array,numeric-method • Cascade</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Methods for selecting genes — geneSelection,micro_array,micro_array,numeric-method" />
<meta property="og:description" content="Selection of differentially expressed genes." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Cascade</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="https://cran.r-project.org/web/packages/Cascade/vignettes/Cascade.pdf">Introduction to the Cascade package and application to the GSE39411 dataset</a>
    </li>
    <li>
      <a href="https://cran.r-project.org/web/packages/Cascade/vignettes/E-MTAB-1475_re-analysis.pdf">Additional application of the Cascade package to E-MTAB-1475 dataset</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fbertran/Cascade">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Methods for selecting genes</h1>
    <small class="dont-index">Source: <a href='https://github.com/fbertran/Cascade/blob/master/R/micro_array.R'><code>R/micro_array.R</code></a></small>
    <div class="hidden name"><code>geneSelection-micro_array-micro_array-numeric-method.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Selection of differentially expressed genes.</p>
    </div>

    <pre class="usage"><span class='co'># S4 method for micro_array,micro_array,numeric</span>
<span class='fu'>geneSelection</span><span class='op'>(</span>
  <span class='va'>x</span>,
  <span class='va'>y</span>,
  <span class='va'>tot.number</span>,
  data_log <span class='op'>=</span> <span class='cn'>TRUE</span>,
  wanted.patterns <span class='op'>=</span> <span class='cn'>NULL</span>,
  forbidden.patterns <span class='op'>=</span> <span class='cn'>NULL</span>,
  peak <span class='op'>=</span> <span class='cn'>NULL</span>,
  alpha <span class='op'>=</span> <span class='fl'>0.05</span>,
  Design <span class='op'>=</span> <span class='cn'>NULL</span>,
  lfc <span class='op'>=</span> <span class='fl'>0</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>either a micro_array object or a list of micro_array objects. In
the first case, the micro_array object represents the stimulated
measurements. In the second case, the control unstimulated data (if present)
should be the first element of the list.</p></td>
    </tr>
    <tr>
      <th>y</th>
      <td><p>either a micro_array object or a list of strings. In the first
case, the micro_array object represents the stimulated measurements. In the
second case, the list is the way to specify the contrast:</p><dl>
<dt>First element:</dt><dd><p>condition, condition&amp;time or pattern. The condition
specification is used when the overall is to compare two conditions.  The
condition&amp;time specification is used when comparing two conditions at two
precise time points. The pattern specification allows to decide which time
point should be differentially expressed.</p></dd> <dt>Second element:</dt><dd><p>a vector
of length 2. The two conditions which should be compared. If a condition is
used as control, it should be the first element of the vector. However, if
this control is not measured throught time, the option cont=TRUE should be
used.</p></dd> <dt>Third element:</dt><dd><p>depends on the first element.  It is no needed
if condition has been specified.  If condition&amp;time has been specified, then
this is a vector containing the time point at which the comparison should be
done. If pattern has been specified, then this is a vector of 0 and 1 of
length T, where T is the number of time points. The time points with desired
differential expression are provided with 1.</p></dd>
</dl></td>
    </tr>
    <tr>
      <th>tot.number</th>
      <td><p>an integer. The number of selected genes. If tot.number &lt;0
all differentially genes are selected. If tot.number &gt; 1, tot.number is the
maximum of diffenrtially genes that will be selected.  If 0&lt;tot.number&lt;1,
tot.number represents the proportion of diffenrentially genes that are
selected.</p></td>
    </tr>
    <tr>
      <th>peak</th>
      <td><p>interger. At which time points measurements should the genes be
selected [optionnal for geneSelection].</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Optional arguments:</p><dl>
<dt>M2</dt><dd><p>a micro_array object.
The unstimulated measurements.</p></dd> <dt>data_log</dt><dd><p>logical (default to TRUE) ;
should data be logged ?</p></dd> <dt>wanted.patterns</dt><dd><p>a matrix with wanted
patterns [only for geneSelection].</p></dd> <dt>forbidden.patterns</dt><dd><p>a matrix with
forbidden patterns [only for geneSelection].</p></dd> <dt>durPeak</dt><dd><p>vector of size
2 (default to c(1,1)) ; the first elements gives the length of the peak at
the left, the second at the right. [only for genePeakSelection]</p></dd>
<dt>abs_val</dt><dd><p>logical (default to TRUE) ; should genes be selected on the
basis of their absolute value expression ? [only for genePeakSelection]</p></dd>
<dt>alpha_diff</dt><dd><p>float ; the risk level</p></dd> 
</dl></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A micro_array object.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Jung, N., Bertrand, F., Bahram, S., Vallat, L., and
Maumy-Bertrand, M. (2014). Cascade: a R-package to study, predict and
simulate the diffusion of a signal through a temporal gene network.
<em>Bioinformatics</em>, btt705.</p>
<p>Vallat, L., Kemper, C. A., Jung, N., Maumy-Bertrand, M., Bertrand, F.,
Meyer, N., ... &amp; Bahram, S. (2013). Reverse-engineering the genetic
circuitry of a cancer cell with predicted intervention in chronic
lymphocytic leukemia. <em>Proceedings of the National Academy of
Sciences</em>, 110(2), 459-464.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Nicolas Jung, Frédéric Bertrand , Myriam Maumy-Bertrand.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Frederic Bertrand, Myriam Maumy-Bertrand.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


